irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -0,0 +1,133 @@
+#include "irods_rest_api_base.h"
+#include "constants.hpp"
+#include "utils.hpp"
+
+#include <irods/filesystem.hpp>
+#include <irods/rodsClient.h>
+#include <irods/irods_random.hpp>
+#include <irods/rodsErrorTable.h>
+#include <irods/irods_at_scope_exit.hpp>
+#include <irods/irods_query.hpp>
+
+#include <pistache/http_headers.h>
+#include <pistache/optional.h>
+#include <pistache/router.h>
+
+namespace irods::rest
+{
+    const std::string service_name { "irods_rest_cpp_logicalpath_server" };
+
+    // this is contractually tied directly to the api implementation
+    class logical_path : public api_base
+    {
+    public:
+        logical_path()
+            : api_base{service_name}
+        {
+            info("Endpoint initialized.");
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        rename_dispatcher(const Pistache::Rest::Request& _request,
```

Comment:
It feels a little cluttered to just write all the specific functions as member functions of this class, especially if we ever need to extend it to functionality that requires a lot of sub-routines to write in a reasonable way. For the first draft it oughtn't be a problem, but at first possible opportunity I'd like to modularize this a little bit.

Reply:
Capture this in an issue in this repository. When that is done, please link here and resolve the comment.

Reply:
Addressed by #122